website/docs/docs/build/custom-schemas.md

```diff
@@ -83,26 +84,63 @@ Under the hood, dbt uses a macro called `generate_schema_name` to determine the
 
 ## Advanced custom schema configuration
 
+You can customize schema name generation in dbt depending on your needs, such as creating a custom macro named `generate_schema_name` in your project or using the built-in macro for environment-based schema names. The built-in macro follows a pattern of generating schema names based on the environment, making it a convenient alternative.
+
+If your dbt project has a macro that’s also named `generate_schema_name`, dbt will always use the macro in your dbt project instead of the default macro.
+
 ### Changing the way dbt generates a schema name
-If your dbt project includes a macro that is also named `generate_schema_name`, dbt will _always use the macro in your dbt project_ instead of the default macro.
 
-Therefore, to change the way dbt generates a schema name, you should add a macro named `generate_schema_name` to your project, where you can then define your own logic.
+To modify how dbt generates schema names, you should add a macro named `generate_schema_name` to your project and customize it according to your needs:
+
+- Copy and paste the `generate_schema_name` macro into a file named 'generate_schema_name'.
+
+- Modify the target schema by either using [target variables](/reference/dbt-jinja-functions/target) or [env_var](/reference/dbt-jinja-functions/env_var). Check out our [Advanced Deployment - Custom Environment and job behavior](https://courses.getdbt.com/courses/advanced-deployment) course video for more details. 
+
+**Note**: dbt will ignore any custom `generate_schema_name` macros included in installed packages. 
+
+<details>
+<summary>❗️ Warning: Don't replace <code>default_schema</code> in the macro.</summary>
 
-Note: dbt ignores any custom `generate_schema_name` macros that are part of a package installed in your project.
+If you're modifying how dbt generates schema names, don't just replace ```{{ default_schema }}_{{ custom_schema_name | trim }}``` with ```{{ custom_schema_name | trim }}``` in the ```generate_schema_name``` macro. 
+
+This prevents developers from overriding each other's models when custom schemas are defined. This also prevents issues during development and continuous integration (CI).
+
+❌ The following code block is an example of what your code _should not_ look like: 
```

Comment:
If we're going to give an example of what they should not do, in the code body should we call out the areas that are incorrect and include text as to why it will cause problems?

Reply:
thanks @matthewshaver , have tweaked this now. can you have a re review and let me know what you think?